eth_utils/address.py

```diff
@@ -6,25 +6,25 @@
     is_hex,
     remove_0x_prefix,
 )
-from .string import (
-    coerce_args_to_text,
-    coerce_args_to_bytes,
-    coerce_return_to_text,
-    coerce_return_to_bytes,
+from .conversions import (
+    to_bytes,
+    to_text,
 )
 from .types import (
+    is_bytes,
     is_string,
 )
 from .formatting import (
     is_prefixed,
 )
 
 
-@coerce_args_to_text
-def is_hex_address(value):
+def is_hex_address(primitive=None, hexstr=None, text=None):
```

Comment:
I think that this should be instead done so that it takes a single `value` parameter, and returns `False` if the value is not a text type (`is_text`).  Also, remove the `@coerce_XXX` decorator.  We can adopt the updated convention that hex addresses **must** be of text type.